Migrate to native label index during store upgrade

MishaDemianenko opened the pull request:
Add additional migrator participant that will create and populate
native label index as part of store upgrade if it does not exist yet.
Migrator is fully integrated into upgrade procedure: it's separate migration step
and support population progress reporting.

MishaDemianenko committed:
Migrate to native label index during store upgrade

Add additional migrator participant that will create and populate
native label index as part of store upgrade if it does not exist yet.
Migrator is fully integrated into upgrade procedure: it's separate migration step
and support population progress reporting.

MishaDemianenko commented:
<b>This PR does not force migration from 3.2 to 3.3 as it is now</b>

chrisvest commented on community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/NativeLabelScanStoreMigrator.java:
```diff
@@ -0,0 +1,135 @@
+        if ( nativeLabelScanStoreMigrated )
+        {
+            File nativeLabelIndex = new File( migrationDir, NativeLabelScanStore.FILE_NAME );
+            fileSystem.moveToDirectory( nativeLabelIndex, storeDir );
```
This won't work on blockdevice. Take a look at how the `StoreMigrator` does it.

chrisvest commented on community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/NativeLabelScanStoreMigrator.java:
```diff
@@ -0,0 +1,135 @@
+
+    private boolean migrateNativeLabelScanStore( File storeDir )
+    {
+        return !fileSystem.fileExists( new File( storeDir, NativeLabelScanStore.FILE_NAME ) );
```
This one is also problematic on blockdevice.

chrisvest reviewed (changes requested):
This would break store migration on blockdevice, and also a relevant integration test is failing.

MishaDemianenko committed:
Add support for mysterious block device that use a file system that is
not a whole db file system, use streams that are not streams and live
in its own world of bytes nibbles and bits

MishaDemianenko commented:
@chrisvest updated
